### PR TITLE
[Manga9] Added Manga9 connector

### LINKED ISSUE #4794: [Site Request] Mangapro
**Name of the website**
Mangapro

**Website urls (examples below)**
- Site: https://mangapro.top/
- Manga List: https://mangapro.top/list/
- Manga example: https://mangapro.top/%e4%b9%99%e5%a5%b3%e3%82%b2%e3%83%bc%e8%bb%a2%e9%80%81%e3%80%81%e4%bf%ba%e3%81%8c%e3%83%92%e3%83%ad%e3%82%a4%e3%83%b3%e3%81%a7%e6%95%91%e4%b8%96%e4%b8%bb-raw-free/

- Chapter online viewer example: https://mangapro.top/chapters/%e3%80%90%e7%ac%ac1%e8%a9%b1%e3%80%91%e4%b9%99%e5%a5%b3%e3%82%b2%e3%83%bc%e8%bb%a2%e9%80%81%e3%80%81%e4%bf%ba%e3%81%8c%e3%83%92%e3%83%ad%e3%82%a4%e3%83%b3%e3%81%a7%e6%95%91%e4%b8%96%e4%b8%bb-raw/

**Languages**
Japanese

Just a side note, if anyone knows, what's the new url (if there is one) for doki1001.com. Since it looks like it's offline

### PR DESCRIPTION
**Pull request by Druzai**
Added connector based on code in https://github.com/manga-download/hakuneko/issues/4794#issuecomment-1187282401
Resolves #4794

### CODE REVIEW

File: src/web/mjs/connectors/Manga9.mjs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        });
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryChapters);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.match(/【[^【】]+】/g).map(m => m.replace(/[【】]/g, '')).pop(),
```
Comment (27136773) [inline src/web/mjs/connectors/Manga9.mjs]: May use simplified regex with matching groups `element.text.match(/【(.+)】\s*$/)[1]`

File: src/web/mjs/connectors/Manga9.mjs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        this.queryMangas = 'div.card-body div.rotate-img a';
+        this.queryChapters = 'div.list-scoll a';
+        this.queryPages = 'div.card-wrap source';
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangaTitle);
+        const id = uri.pathname;
+        const title = data[0].textContent.replace('(Raw – Free)', '').trim();
```
Comment (27136773) [inline src/web/mjs/connectors/Manga9.mjs]: May use regex in case there are chapters witout `Free` keyword `/\(Raw( - Free)?\)/`

**Review by 27136773 — APPROVED** [review]
Looks good, regex may be optimized

File: src/web/mjs/connectors/Manga9.mjs (hunk 1 above)
Comment (Druzai) [reply src/web/mjs/connectors/Manga9.mjs]: Fixed

File: src/web/mjs/connectors/Manga9.mjs (hunk 2 above)
Comment (Druzai) [reply src/web/mjs/connectors/Manga9.mjs]: Fixed

**Pull request merged**